robot-db/src/main/java/dbconfiguration/SpringDBConfiguration.java

```diff
@@ -46,20 +52,30 @@ private Properties hibernateProperties() {
         return properties;
     }
 
+    /**
+     * Entity manager factory bean.
+     * @return - new factory bean of LocalContainerEntityManager
+     */
     @Bean
-    LocalSessionFactoryBean sessionFactory() {
-        LocalSessionFactoryBean sessionFactory = new LocalSessionFactoryBean();
-        sessionFactory.setDataSource(dataSource());
-        sessionFactory.setPackagesToScan("domain");
-        sessionFactory.setHibernateProperties(hibernateProperties());
-        return sessionFactory;
+    public LocalContainerEntityManagerFactoryBean entityManagerFactory() {
+        LocalContainerEntityManagerFactoryBean entityManagerFactory = new LocalContainerEntityManagerFactoryBean();
+        HibernateJpaVendorAdapter adapter = new HibernateJpaVendorAdapter();
+        adapter.setDatabase(Database.POSTGRESQL);
+        entityManagerFactory.setJpaVendorAdapter(adapter);
+        entityManagerFactory.setDataSource(dataSource());
+        entityManagerFactory.setPackagesToScan("domain");
```

Comment:
Very good candidate for an external property. If there ever is a time where you want to change packages scanned for beans to something else, it will be easy to do, and may not necessarily require restarting every service.